src/cfnlint/custom_rules.py

```diff
@@ -4,37 +4,42 @@
 """
 # pylint: disable=W0108
 # pylint: disable=W0622
+
 import logging
 import cfnlint
+import cfnlint.runner
 import cfnlint.customRules.Operators
 import cfnlint.customRules.Rule
 
 LOGGER = logging.getLogger(__name__)
-Operator = {'EQUALS': lambda x, y: cfnlint.customRules.Operators.equalsOp(x, y),
-            'PLACEHOLDER': lambda x, y: LOGGER.debug('Placeholder Op')}
+Operator = {'EQUALS': lambda x, y, z: cfnlint.customRules.Operators.equalsOp(x, y, z),
+            'NOT_EQUALS': lambda x, y, z: cfnlint.customRules.Operators.notEqualsOp(x, y, z),
+            'INSET': lambda x, y, z: cfnlint.customRules.Operators.InSetOp(x, y, z),
+            'NOT_INSET': lambda x, y, z: cfnlint.customRules.Operators.NotInSetOp(x, y, z)}
 
-def check(filename, template):
+def check(filename, template, rules, runner):
     """ Process custom rule file """
     matches = []
     with open(filename) as customRules:
         line_number = 1
         for line in customRules:
             LOGGER.debug('Processing Custom Rule Line %d', line_number)
             line = line.replace('"', '')
-            rule = cfnlint.customRules.Rule.make_rule(line.split(' '))
+            rule = cfnlint.customRules.Rule.make_rule(line, line_number)
             if rule.valid and rule.resourceType[0] != '#':
                 try:
                     resource_properties = template.get_resource_properties([rule.resourceType])
-                    result = Operator[rule.operator](rule, resource_properties)
-                    if result != rule.operator:
-                        matches.append(cfnlint.rules.Match(line_number, '0', '0', '0', filename, CustomRule('E9999'), result, None))
+                    operator_result = Operator[rule.operator](template, rule, resource_properties)
+                    matches += operator_result
                 except KeyError:
-                    matches.append(cfnlint.rules.Match(line_number, '0', '0', '0', filename, CustomRule('E9999'),
-                                                       str(rule.operator) + ' not in supported operators: [EQUALS] at ' + str(line), None))
+                    matches.append(cfnlint.rules.Match(
+                        1, 1,
+                        1, 1,
+                        template.filename, cfnlint.customRules.Operators.CustomRule('E9999'),
+                        str(rule.operator) + ' not in supported operators: [EQUALS] at ' + str(line), None))
```

Comment:
Outdated, marking to remind me to fix this.

Reply:
Shouldn't be hard coded, instead get list of dict keys and print them here so it dynamically updates